### PR TITLE
Cypress testing tch4lla

### LINKED ISSUE #168: Start Cypress Tests for frontend
# Requirements

- [x] cypress tooling and docs in place
- [x] Skeleton tests
- [x] Finish the skeleton tests in `home.cy.ts`

Please break this work into small parts and share intermediate progress via PR for feedback.

# Dependencies

- None (Add any issue dependencies here)

### PR DESCRIPTION
**Pull request by Tch4lla**
Resolves #168 

# What changed

- Added all testing queries using cypress
- Added data-testid to multiple elements in search bar and footer
- "changed linebreak-style": 0 in eslintrc under rules in order to tackle the linting issue

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ module.exports = {
         ],
         'key-spacing': ['error', { beforeColon: false, afterColon: true }],
         'keyword-spacing': ['error', { before: true, after: true }],
-        'linebreak-style': ['error', 'unix'], // no carriage returns
+        'linebreak-style': 0, // no carriage returns
```
Comment (ann-kilzer) [inline .eslintrc.js:69]: Please don't change this, we do not want carriage returns. Make sure your editor is correctly configured with editorConfig

File: components/Footer.vue [hunk 2]
```diff
@@ -1,34 +1,38 @@
 <template>
     <div class="footer flex border-t-4 border-secondary-bg/40 bg-secondary-bg text-sm
-    text-primary-text font-semibold py-2">
+        text-primary-text font-semibold py-2">
```
Comment (ann-kilzer) [inline components/Footer.vue:3]: Seems like an unrelated change?

**Review by ann-kilzer — CHANGES_REQUESTED** [review]
Thanks for starting this. See comments

File: .eslintrc.js (hunk 1 above)
Comment (Tch4lla) [reply .eslintrc.js:69]: I understand. In regards to this change, I've changed my end of line character to \n so that it is for the unix type, and the linter will return 1100+ errors in regards to the spacing. yarn lint --fix does not autofix anything, neither does the selction 'fix all line space errors'. 

Would you be able to offer any suggestions?

File: components/Footer.vue (hunk 2 above)
Comment (Tch4lla) [reply components/Footer.vue:3]: This probably was a lint format fix.

File: .eslintrc.js (hunk 1 above)
Comment (Tch4lla) [reply .eslintrc.js:69]: linter

I just changed it back to that error unix property and ran yarn lint --fix command

Comment (Tch4lla) [reply .eslintrc.js]: I manually fixed all the errors. The linting should pass at least. Still, I'm not sure why the auto fix functions don't work...

File: components/Footer.vue (hunk 2 above)
Comment (ann-kilzer) [reply components/Footer.vue:3]: As a best practice, we should be separating linting fixes from logical changes. The files have been linted before and auto-checked as part of CI/CD. Unrelated changes distract from the core purpose of this PR: cypress testing

If you confirm that there is an error with the linter (can test by running against `main`), feel free to open a separate PR to address the issue.

File: cypress/e2e/home.cy.ts [hunk 3]
```diff
@@ -10,34 +13,76 @@ describe('Visits the home page', () => {
...
+    it('can select "Japanese" from the language bar', () => {
+        cy.get('[data-testid="search-bar-language"]').should('be.visible')
+
+        cy.get('[data-testid="search-bar-language"]').select('Japanese')
+
+        cy.get('[data-testid="search-bar-language"]').should('have.value', 'Japanese')
+    })
 
+    it('shows doctors nearby', () => {
+        cy.contains('Doctors Nearby').should('be.visible')
```
Comment (ann-kilzer) [inline cypress/e2e/home.cy.ts:50]: Can we extend this one to ensure that the expected content is in the component?

File: cypress/e2e/home.cy.ts [hunk 4]
```diff
@@ -10,34 +13,76 @@ describe('Visits the home page', () => {
...
+        })
 
         // verify Netlify citation
+        it('navigates to the netlify page', () => {
+            cy.get('[data-testid="netlify-link"]').should('be.visible')
+            cy.get('[data-testid="netlify-link"]').should('have.attr', 'href', netlify)
+        })
 
         // privacy
+        it('naviagtes to private policy', () => {
```
Comment (ann-kilzer) [inline cypress/e2e/home.cy.ts:75]: looking good 👍🏻

File: components/TopNav.vue [hunk 5]
```diff
@@ -10,7 +10,7 @@
                 </svg>
                 <div class="title-text flex flex-col" data-testid="logo">
                     <div class="text-lg text-primary group-hover:text-primary-hover">
-                        Find a Doc 
+                        Find a Doc
```
Comment (ann-kilzer) [inline components/TopNav.vue:13]: 👀 looks like this one WAS unlinted 🤔

File: cypress/e2e/home.cy.ts [hunk 6]
```diff
@@ -10,34 +13,76 @@ describe('Visits the home page', () => {
...
 
-    it.skip('displays the footer', () => {
+    describe('Checks footer links', () => {
+        it('navigates to github', () => {
         // TODO
 
-        // verify link to GitHub
+            // verify link to GitHub
+            cy.get('[data-testid="github-link"]').should('be.visible')
+            cy.get('[data-testid="github-link"]').should('have.attr', 'href', 'https://github.com/ourjapanlife/findadoc-web/')
```
Comment (ann-kilzer) [inline cypress/e2e/home.cy.ts:59]: Should we make a const for the github url, for consistency?

File: cypress/e2e/home.cy.ts [hunk 7]
```diff
@@ -10,34 +13,76 @@ describe('Visits the home page', () => {
...
+    })
+
     it('allows setting search fields', () => {
         cy.get('[data-testid="search-button"]').should('be.visible')
         // TODO
     })
 
-    it.skip('selects a language', () => {
+    it('can select "select a language"', () => {
+        cy.get('[data-testid="search-bar-language"]').trigger('click')
```
Comment (ann-kilzer) [inline cypress/e2e/home.cy.ts:29]: Let's add some assertions to all the tests (or skip them if we don't have enough of the homepage working to finish)

**Review by ann-kilzer — COMMENTED** [review]
Took a closer look at the tests and they're looking on the right track

File: .eslintrc.js (hunk 1 above)
Comment (ann-kilzer) [reply .eslintrc.js]: On our readme we note that everyone should be using the https://editorconfig.org/ tool, and if you use VSCode, you will need to install a special plugin. This file will set up expected line endings and formatting

🤔 Can we make this important setup step clearer somehow? Maybe we need to highlight it in CONTRIBUTING.md

Git on windows sometimes does sneaky things, replacing `\n` with `CRLF` on checkout, and converting back on upload.

File: cypress/e2e/home.cy.ts (hunk 6 above)
Comment (Tch4lla) [reply cypress/e2e/home.cy.ts:59]: I'll take care of that

File: cypress/e2e/home.cy.ts (hunk 7 above)
Comment (Tch4lla) [reply cypress/e2e/home.cy.ts:29]: I understand. I'll skip this one. I guess in the following tests if I can select the languages, then this test by default should pass. I'll keep that in mind as I go forward

File: components/TopNav.vue (hunk 5 above)
Comment (Tch4lla) [reply components/TopNav.vue:13]: This might be the lf vs crlf line ending problem I was taking about?

File: .eslintrc.js (hunk 1 above)
Comment (Tch4lla) [reply .eslintrc.js]: Hmm, I have the editorconfig extension installed on my vs code and I have the editorconfig file pulled. That is interesting about the \n for git on windows.

File: components/Footer.vue (hunk 2 above)
Comment (Tch4lla) [reply components/Footer.vue:3]: Thank you for that. Previously when I was working on the other PR for the results store, I was always running into the linting issue, and was advised to run yarn lint fix, so I assumed that what I should always do before a push. I'll keep that in mind for the future

**Review by ann-kilzer — APPROVED** [review]
Going to remove my blocking "changes requested" as the line ending has been fixed. Tests are on the right track. There are a few suggested changes I've already left, but this is a good step forward. Thank you!
